components/kyma-environment-broker/internal/process/deprovisioning/remove_runtime.go

```diff
@@ -53,8 +53,12 @@ func (s *RemoveRuntimeStep) Run(operation internal.DeprovisioningOperation, log
 	}
 
 	if instance.RuntimeID == "" {
-		log.Warn("Runtime not exist")
-		return operation, 0, nil
+		// happens when provisioning process failed and Create_Runtime step was never reached
+		log.Warnf("Runtime does not exist for instance id %q", instance.InstanceID)
+
+		operation, _, _ := s.operationManager.OperationSucceeded(operation, "Runtime was never provisioned")
+		// return repeat mode (1 sec) to start the initialization step which will finish process and remove instance
+		return operation, 1 * time.Second, nil
```

Comment:
```golang
func (om *DeprovisionOperationManager) OperationSucceeded(operation internal.DeprovisioningOperation, description string) (internal.DeprovisioningOperation, time.Duration, error) {
	updatedOperation, repeat := om.update(operation, domain.Succeeded, description)
	// repeat in case of storage error
	if repeat != 0 {
		return updatedOperation, repeat, nil
	}

	return updatedOperation, 0, nil
}
```

It never returns error, in case of any problem operation is returned with some time value !=0 causing the process repetition. In this case we are doing exactly the same thing with:
```golang
return operation, 1 * time.Second, nil
```